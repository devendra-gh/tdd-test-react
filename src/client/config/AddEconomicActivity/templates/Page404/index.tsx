import React from 'react';
import { withTemplateHooks, IVariables } from '@tamm/app-composer';
import routePropTypes from '@tamm/app-composer/client/propTypes/route';
import PageNotFoundTemplate from '@tamm/ui-lib-v2-page-not-found-template';
import Container from 'client/containers';

function Page404Template(props: IVariables) {
  return (
    <Container>
      <PageNotFoundTemplate
        backButton={{
          label: props.i18n(props.label),
          onClick: () => props.history.push(props.btnBack),
        }}
        i18n={props.i18n}
      />
    </Container>
  );
}

Page404Template.propTypes = {
  ...routePropTypes,
};

export default withTemplateHooks(Page404Template);
