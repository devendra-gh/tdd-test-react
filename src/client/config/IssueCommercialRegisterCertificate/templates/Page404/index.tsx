import React from 'react';
import { withTemplateHooks, IVariables } from '@tamm/app-composer';
import routePropTypes from '@tamm/app-composer/client/propTypes/route';
import PageNotFoundTemplate from '@tamm/ui-lib-v2-page-not-found-template';
import Container from 'client/containers';

function Page404(props: IVariables) {
  return (
    <Container>
      <div role="navigation" arial-label="Page Not Found">
        <PageNotFoundTemplate
          backButton={{
            'aria-label': props.i18n(props.label),
            label: props.i18n(props.label),
            onClick: () => props.history.push(props.btnBack),
          }}
          i18n={props.i18n}
        />
      </div>
    </Container>
  );
}

Page404.propTypes = {
  ...routePropTypes,
};

export default withTemplateHooks(Page404);
