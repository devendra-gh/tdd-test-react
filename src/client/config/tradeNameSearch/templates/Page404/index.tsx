import React from 'react';
import { withTemplateHooks, IVariables } from '@tamm/app-composer';
import routePropTypes from '@tamm/app-composer/client/propTypes/route';
import PageNotFoundTemplate from '@tamm/ui-lib-v2-page-not-found-template';
import Container from 'client/containers';

function Page404(props: IVariables) {
  return (
    <Container>
      <PageNotFoundTemplate
        backButton={{
          label: props.i18n(props.label),
          onClick: () => {
            props.history.push(props.btnBack);
          },
        }}
        i18n={props.i18n}
      />
    </Container>
  );
}

Page404.propTypes = {
  ...routePropTypes,
};

export default withTemplateHooks(Page404);
