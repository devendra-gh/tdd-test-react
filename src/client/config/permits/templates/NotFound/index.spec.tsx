import React, { FunctionComponent } from 'react';
import { shallow } from 'enzyme';
import Page404 from './index';

jest.mock('@tamm/app-composer', () => ({
  withTemplateHooks: (component: FunctionComponent) => component,
}));
// eslint-disable-next-line no-console
console.log = () => {};
console.info = () => {};
console.warn = () => {};
console.error = () => {};

describe('renew-licence/templates/Login', () => {
  let props: any;

  beforeEach(() => {
    props = {
      match: {},
      location: {},
      // i18n: (key: string) => key,
      onClick: jest.fn(),
      history: {
        push: jest.fn(),
      },
      i18n: jest.fn(),
    };
  });

  // afterEach(cleanup);

  it('should render successfully', () => {
    const component = shallow(<Page404 {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('should cover i18n property', () => {
    const component = shallow(<Page404 {...props} />);
    const template = component.find('LocalizedComponent(PageNotFoundTemplate)');
    const allProps: any = template.props();
    const { i18n } = allProps;
    expect(i18n()).toBeFalsy();
  });
  // eslint-disable-next-line no-console
  console.log = () => {};
  console.info = () => {};
  console.warn = () => {};
  console.error = () => {};

  describe('BackButton', () => {
    it('should handle button click', () => {
      const component = shallow(<Page404 {...props} />);
      const template = component.find(
        'LocalizedComponent(PageNotFoundTemplate)',
      );
      const allProps: any = template.props();
      const { backButton } = allProps;
      backButton.onClick();
      expect(props.history.push).toBeCalled();
    });
  });
});
