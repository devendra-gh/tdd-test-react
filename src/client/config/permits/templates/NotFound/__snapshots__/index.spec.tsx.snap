// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renew-licence/templates/Login should render successfully 1`] = `
<Container>
  <LocalizedComponent(PageNotFoundTemplate)
    backButton={
      Object {
        "label": undefined,
        "onClick": [Function],
      }
    }
    i18n={[Function]}
  />
</Container>
`;
